apps/api/v2/src/modules/organizations/controllers/event-types/organizations-event-types.controller.ts

```diff
@@ -29,40 +33,61 @@ import {
   HttpStatus,
   NotFoundException,
   Query,
+  UseInterceptors,
 } from "@nestjs/common";
 import { ApiTags as DocsTags } from "@nestjs/swagger";
 
-import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import { ERROR_STATUS, SUCCESS_STATUS } from "@calcom/platform-constants";
 import {
   CreateTeamEventTypeInput_2024_06_14,
   GetTeamEventTypesQuery_2024_06_14,
   SkipTakePagination,
   UpdateTeamEventTypeInput_2024_06_14,
 } from "@calcom/platform-types";
 
+export type EventTypeHandlerResponse = {
+  data: DatabaseTeamEventType[] | DatabaseTeamEventType;
+  status: typeof SUCCESS_STATUS | typeof ERROR_STATUS;
+};
+
 @Controller({
   path: "/v2/organizations/:orgId",
   version: API_VERSIONS_VALUES,
 })
 @DocsTags("Organizations Event Types")
 export class OrganizationsEventTypesController {
-  constructor(private readonly organizationsEventTypesService: OrganizationsEventTypesService) {}
+  constructor(
+    private readonly organizationsEventTypesService: OrganizationsEventTypesService,
+    private readonly inputService: InputOrganizationsEventTypesService,
+    private readonly inputUserEventTypesService: InputEventTypesService_2024_06_14
+  ) {}
 
   @Roles("TEAM_ADMIN")
   @PlatformPlan("ESSENTIALS")
   @UseGuards(ApiAuthGuard, IsOrgGuard, RolesGuard, IsTeamInOrg, PlatformPlanGuard, IsAdminAPIEnabledGuard)
+  @UseInterceptors(OutputEventTypesResponseInterceptor<CreateTeamEventTypeOutput>)
   @Post("/teams/:teamId/event-types")
   async createTeamEventType(
     @GetUser() user: UserWithProfile,
     @Param("teamId", ParseIntPipe) teamId: number,
     @Param("orgId", ParseIntPipe) orgId: number,
     @Body() bodyEventType: CreateTeamEventTypeInput_2024_06_14
-  ): Promise<CreateTeamEventTypeOutput> {
+  ): Promise<EventTypeHandlerResponse> {
+    await this.inputService.validateHosts(teamId, bodyEventType.hosts);
```

Comment:
Why not have a function in the input service to validate hosts, transform body and validate event type inputs and then return transformed body only that is passed to create team event type.

Reply:
If we could figure out how to pass `teamId` to the pipe, we could convert this into a pipe that transforms the data and then validates it.

would make the code much cleaner

see https://github.com/calcom/cal.com/pull/16314/files/9926a012f4c45596734197c1d2e6a83d462bcf06#r1743564611